src/Substructure.jl

```diff
@@ -0,0 +1,304 @@
+"""
+    has_parents(p, clusterseq) -> (boolean, Int64, Int64)
+
+Checks if the jet `p` is a child of two other jets, after clustering 
+
+# Arguments
+- `p`: The jet to check.
+- `clusterseq`: The cluster sequence object.
+
+# Returns
+- (boolean, Int64, Int64): true or false depending on if the jet has a parent or not. If the jet has a parent, returns the indices of the parent jets in the history element. Otherwise, returns -2 (NonexistentParent).
+"""
+function has_parents(p::PseudoJet, clusterseq::ClusterSequence)
+    history = clusterseq.history
+    N = p._cluster_hist_index
+    p1 = history[N].parent1
+    p2 = history[N].parent2
+
+    p1 == p2 == NonexistentParent ? result = false : result = true
+    return (result, p1, p2)
+end
+
+"""
+    deltaR(jet1, jet2) -> Float64
+
+Function to calculate the distance in the y-ϕ plane between two jets `jet1` and `jet2`
+
+# Arguments
+- `jet1`: The first jet.
+- `jet2`: The second jet.
+
+# Returns
+- Float64: The Euclidean distance in the y-ϕ plane for the two jets.
+"""
+function deltaR(jet1::PseudoJet, jet2::PseudoJet)
+    y1, phi1 = rapidity(jet1), phi(jet1)
+    y2, phi2 = rapidity(jet2), phi(jet2)
+
+    d_y = y1 - y2
+    d_phi = phi1 - phi2
+    d_phi = abs(d_phi) > π ? 2π - abs(d_phi) : d_phi
+
+    return sqrt(d_y^2 + d_phi^2)
+end
+
+# Function to calculate kt distance between two pseudojets
+function kt_distance(jet1::PseudoJet, jet2::PseudoJet)
+    p1 = pt2(jet1)
+    p2 = pt2(jet2)
+
+    d_R = deltaR(jet1, jet2)
+    return min(p1, p2) * d_R^2
+end
+
+"""
+    recluster(jet, clusterseq, rad, mtd) -> ClusterSequence
+
+Reclusters the constituents of a given jet `jet` with a different clustering method `mtd` and different jet radius `rad`.
+
+# Arguments
+- `jet`: The jet whose constituents are to be reclustered.
+- `clusterseq`: The cluster sequence from which the original jet is obtained.
+- `rad`: The new jet radius.
+- `mtd`: The new clustering method.
+
+# Returns
+- ClusterSequence: The new cluster sequence.
+"""
+function recluster(jet::PseudoJet, clusterseq::ClusterSequence; R = 1.0,
+                   algorithm = JetAlgorithm.CA)
+    cons = constituents(jet, clusterseq)
+    new_clusterseq = jet_reconstruct(cons; p = nothing, R = R, algorithm = algorithm,
+                                     strategy = RecoStrategy.Best)
+
+    return new_clusterseq
+end
+
+function sort_jets!(event_jet_array)
```

Comment:
I still find the need for this function questionable. You can just replace it with 

`sort!(event_jet_array, by = pt2, rev = true)`

so it's really not needed.